### PR TITLE
Fix support for Petrified Blood

### PR DESCRIPTION
**Pull request by Wires77**
The breakdown for the added life cost isn't in there, but this seems to work.  There might be a better way to implement this, just hacked it together quick over lunch.

### CODE REVIEW

**Review by ppoelzl — APPROVED** [review]
It's alright as it is.

**Review by PJacek — CHANGES_REQUESTED** [review]
This check seems to fail https://github.com/PathOfBuildingCommunity/PathOfBuilding/blob/778b4a588d7d0364e07d1535f9cb6a8d780234c4/src/Modules/CalcOffence.lua#L1006 and displays the breakdown for equal base and final cost. If you change the output formatting to float you can see that the values are actually different
cost
According to my testing (6 base mana cost with 1.76 multiplier results in 4 life cost not 3) the value coming from conversion is used in calculations without rounding so we should probably display fractions.

For the breakdown you can just add `ManaCostAsLifeCost` to the listed mods

The aura totem costs weren't working perfectly before but now are completely broken. Moving some of the calculations to earlier pass misses the base cost added by summoning and the code refactoring both duplicates flat reservation as percent cost and misses percent reservation.
Anyway, I don't think anyone uses this interaction so it can stay like that. I can fix it later.

File: src/Modules/CalcOffence.lua [hunk 1]
```diff
@@ -977,35 +989,23 @@ function calcs.offence(env, actor, activeSkill)
...
 					if reservedPercent ~= 0 then
-						skillModList:NewMod("LifePercentCostBase", "BASE", reservedPercent, "Totem Reservation")
+						skillModList:NewMod(resource.."PercentCostBase", "BASE", reservedPercent, "Totem Reservation")
 					end
 				end
-				if breakdown and output[resource.."Cost"] ~= cost then
+				if breakdown and output[resource.."Cost"] ~= baseCost then
 					breakdown[resource.."Cost"] = {
-						s_format("%d"..(percent and "%%" or "").." ^8(base "..name.." cost)", cost)
+						s_format("%d"..(percent and "%%" or "").." ^8(base "..name.." cost)", baseCost)
```
[the remark this answers is not in the document]
Comment (ppoelzl) [reply src/Modules/CalcOffence.lua:1008]: Shouldn't this round up to the next integer then? As you can't spend exactly 2.4 life anyways.

[the remark this answers is not in the document]
Comment (PJacek) [reply src/Modules/CalcOffence.lua:1008]: The math is a bit complicated
https://github.com/PathOfBuildingCommunity/PathOfBuilding/blob/778b4a588d7d0364e07d1535f9cb6a8d780234c4/src/Modules/CalcOffence.lua#L994-L997
My point is that the base cost from conversion seems to allow fractions and the calculation uses fractions so we could display it like that

**Pull request merged**